examples/cpp/makefile

```diff
@@ -0,0 +1,22 @@
+CXX = g++
+LD = $(CXX)
+
+OPTIM = -O0
+CXXFLAGS += -Wall -pedantic -std=c++17 $(OPTIM) 
+
+INCLUDE=-I ../../src
+
+LIB= -L ../../lib -lSHTOOLS -lgfortran \
+     -ldl -lfftw3 -lm -llapack -lblas \
```

Comment:
I am ok with this for now, but we might want to consider getting rid of the relative locations of `../../src` and `../../lib` for the case where these have been moved elsewhere (this was an issue with macports/homebrew...). What I have done for the Fortran examples is the following:

In the main Makefile, I created targets like `fortran-tests`, and then called the Makefile in the examples folder using

```
@$(MAKE) -C $(FEXDIR) -f Makefile all F95="$(F95)" F95FLAGS="$(F95FLAGS)" LIBNAME="$(LIBNAME)" FFTW="$(FFTW)" LAPACK="$(LAPACK)" BLAS="$(BLAS)" LIBPATH="$(LIBPATH)" MODFLAG="$(MODFLAG)"
```

Its ugly, but it works. The good thing about this approach is that the main Makefile defines all of these variables once (some of which might be different for different compilers). The bad part is that the file `examples/cpp/Makefile` can't be used by itself (unless you set default values for all the variables, which are compiler and architecture dependent).

In any case, it would be useful to include a simple `cpp-tests` target in the main Makefile so that it could be easily run on Travis.

Reply:
another way to deal with this could be creating 'machine files', for example, one called `linux-gfortran`:

```make
# Default gfortran flags
F95FLAGS ?= -m64 -fPIC -O3 -std=gnu -ffast-math
# -march=native
MODFLAG = -I$(MODPATH)
SYSMODFLAG = -I$(SYSMODPATH)
OPENMPFLAGS ?= -fopenmp
BLAS ?= -lblas
LAPACK ?= -llapack
```

include the machine file in the main makefile with

`include $(MAKE_MACHINE)`

and call make with

`MAKE_MACHINE=linux-gfortran make`

the sub makefiles can also include the corresponding file